# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLActiveInfo.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGLActiveInfo.cpp
--- a/dom/canvas/WebGLActiveInfo.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGLActiveInfo.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -146,12 +146,10 @@
   }
 }
 
-WebGLActiveInfo::WebGLActiveInfo(WebGLContext* webgl, GLint elemCount,
-                                 GLenum elemType, bool isArray,
+WebGLActiveInfo::WebGLActiveInfo(GLint elemCount, GLenum elemType, bool isArray,
                                  const nsACString& baseUserName,
                                  const nsACString& baseMappedName)
-    : mWebGL(webgl),
-      mElemCount(elemCount),
+    : mElemCount(elemCount),
       mElemType(elemType),
       mBaseUserName(baseUserName),
       mIsArray(isArray),
@@ -159,6 +157,15 @@
       mBaseMappedName(baseMappedName),
       mBaseType(ElemBaseType(mElemType)) {}
 
+WebGLActiveInfo::WebGLActiveInfo(const WebGLActiveInfo& aOther)
+    : mElemCount(aOther.mElemCount),
+      mElemType(aOther.mElemType),
+      mBaseUserName(aOther.mBaseUserName),
+      mIsArray(aOther.mIsArray),
+      mElemSize(aOther.mElemSize),
+      mBaseMappedName(aOther.mBaseMappedName),
+      mBaseType(aOther.mBaseType) {}
+
 bool WebGLActiveInfo::IsSampler() const {
   switch (mElemType) {
     case LOCAL_GL_SAMPLER_2D:
@@ -185,14 +192,14 @@
 
 ////////////////////////////////////////////////////////////////////////////////
 
-JSObject* WebGLActiveInfo::WrapObject(JSContext* js,
-                                      JS::Handle<JSObject*> givenProto) {
+JSObject* ClientWebGLActiveInfo::WrapObject(JSContext* js,
+                                            JS::Handle<JSObject*> givenProto) {
   return dom::WebGLActiveInfo_Binding::Wrap(js, this, givenProto);
 }
 
-NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE_0(WebGLActiveInfo)
+NS_IMPL_CYCLE_COLLECTION_WRAPPERCACHE_0(ClientWebGLActiveInfo)
 
-NS_IMPL_CYCLE_COLLECTION_ROOT_NATIVE(WebGLActiveInfo, AddRef)
-NS_IMPL_CYCLE_COLLECTION_UNROOT_NATIVE(WebGLActiveInfo, Release)
+NS_IMPL_CYCLE_COLLECTION_ROOT_NATIVE(ClientWebGLActiveInfo, AddRef)
+NS_IMPL_CYCLE_COLLECTION_UNROOT_NATIVE(ClientWebGLActiveInfo, Release)
 
 }  // namespace mozilla